public/react-services/error-management/error-handler/hooks/useErrorHandler.ts

```diff
@@ -0,0 +1,18 @@
+import { ErrorHandler } from '../error-handler';
+
+/**
+ *
+ * @param callback
+ * @returns
+ */
+export const useErrorHandler = async (callback: Function) => {
+  try {
+    let res = await callback();
+    return [res, null];
+  } catch (error) {
+    if (error instanceof Error) {
+      error = ErrorHandler.handleError(error);
+    }
+    return [null, error];
+  }
+};
```

Comment:
**thought**: If this is a ReactJS hook, there is something that is missing, the hook can't return asynchronously. For another hand, and ignoring the implementatin problem, it only covers a use case. Could you clarify what is the intention of this hook?

Reply:
That's right. I've changed this implementation to avoid repeating the error orchestrator call.

Reply:
The use case for this hook is using it to wrap the async calls and avoid the try-catch block for every place. Also, clarify the error catching.
**For instance:** If we wrap a method that calls to the API, we can treat the error for this specific case because the hook returns the error. Right now, we are wrapping many methods within a try-catch block and we don't know which of these methods is throwing the error and can't treat it as separate.